src/tracer/ProcessInfo.cc

```diff
@@ -0,0 +1,47 @@
+#include "ProcessInfo.h"
+
+#include "common/Assert.h"
+
+namespace s2j {
+namespace tracer {
+
+ProcessInfo::ProcessInfo(pid_t pid, std::shared_ptr<ProcessInfo> parent)
+        : pid_{pid}, parent_{parent} {
+    if (parent == nullptr)
+        wholeTreeInfo_ =
+                std::make_shared<std::map<pid_t, std::weak_ptr<ProcessInfo>>>();
+    else
+        wholeTreeInfo_ = parent->wholeTreeInfo_;
+}
+
+std::shared_ptr<ProcessInfo> ProcessInfo::makeProcessInfo(
+        pid_t pid,
+        std::shared_ptr<ProcessInfo> parent) {
+    auto process = std::make_shared<ProcessInfo>(pid, parent);
+
+    auto selfInsert =
+            process->wholeTreeInfo_->emplace(pid, process->shared_from_this());
```

Comment:
Czy tutaj trzeba `shared_from_this` ?

Reply:
Jasne, to najpierw było w konstruktorze ale niezbyt to wtedy działa, więc przeniosłem